AmplifyPlugins/DataStore/Sources/AWSDataStorePlugin/Subscribe/DataStoreObserveQueryOperation.swift

```diff
@@ -230,6 +236,14 @@ class ObserveQueryTaskRunner<M: Model>: InternalTaskRunner, InternalTaskAsyncThr
                       receiveValue: self.onItemChangeAfterSync(mutationEvent:))
         }
     }
+    
+    func subscribeToModelSyncedEvent() {
+        modelSyncedEventSink = Amplify.Hub.publisher(for: .dataStore).sink { event in
+            if event.eventName == HubPayload.EventName.DataStore.modelSynced, case let modelSyncedEvent = event.data as? ModelSyncedEvent, modelSyncedEvent?.modelName == self.modelSchema.name {
+                self.sendSnapshot()
+            }
```

Comment:
does `case` do anything here? I can see `modelSyncedEvent` is of type optional ModelSyncedEvent, i believe we want this:
```
func subscribeToModelSyncedEvent() {
        modelSyncedEventSink = Amplify.Hub.publisher(for: .dataStore).sink { event in
            if event.eventName == HubPayload.EventName.DataStore.modelSynced,
                let modelSyncedEvent = event.data as? ModelSyncedEvent,
                modelSyncedEvent.modelName == self.modelSchema.name {
                self.sendSnapshot()
            }
        }
    }
```
(please keep the line length reasonably within the column guidance (120)